proto/internal/temporal/server/api/historyservice/v1/request_response.proto

```diff
@@ -916,3 +917,24 @@ message CompleteNexusOperationRequest {
 
 message CompleteNexusOperationResponse {
 }
+
+message InvokeStateMachineMethodRequest {
+    // TODO(Tianyu): This is the same as NexusOperationsCompletion but obviously is not about Nexus. This is because
+    // State machine signaling is a generalization of the Nexus mechanisms. Perhaps eventually they should be merged.
+    // Namespace UUID.
+    string namespace_id = 1;
+    // Workflow ID.
+    string workflow_id = 2;
+    // Run ID at the time this token was generated.
+    string run_id = 3;
+    // Reference including the path to the backing Operation state machine and a version + transition count for
+    // staleness checks.
+    temporal.server.api.persistence.v1.StateMachineRef ref = 4;
+
+    string method_name = 5;
+    bytes input = 6;
```

Comment:
Maybe put a comment about how this will be deserialized? And one on the output field of the response?